File: src/kipper/effects/SampleConfigImpl.py
from kipper.effects import ParticleEmitterConfig
from kipper.effects.transitions import Linear

import math
from random import random

L = Linear()

class SampleConfigImpl(ParticleEmitterConfig):

    def getMaxParticles(self):
        return 48

    def getDurationTicks(self):
        return 45

    def getSpawnRate(self):
        return 1

    def isContinuous(self):
        return False

    def isRectShape(self, p):
        return False

    def getSize(self, p):
        return int(L.call(p.ticks, 15, -15, self.getDurationTicks()))

    def getTheta(self, p):
        if p.ticks < 1:
            return random() * 2 * math.pi
        return p.theta

    def getSpeed(self, p):
        if p.ticks < 1:
            return random() * 2
        return p.speed

    def getHue(self, p):
        return 60.0 / 360.0

    def getSaturation(self, p):
        return 1

    def getBrightness(self, p):
        return 1

File: src/kipper/effects/SampleConfigImpl.java
package kipper.effects;

import kipper.effects.transitions.*;

public class SampleConfigImpl extends ParticleEmitterConfig
{
    @Override public int getMaxParticles() { return 10; }
    @Override public int getDurationTicks() { return 50; }
    @Override public double getSpawnRate() { return 1; }
    @Override public boolean isContinuous() { return false; }
    @Override public boolean isRectShape(Particle p) { return false; }
    @Override public int getSize(Particle p) { return (int)(new EaseOutCubic().call(p.ticks, 15, -15, getDurationTicks())); }
    @Override public double getTheta(Particle p) { return p.ticks < 1 ? Math.random() * 2 * Math.PI : p.theta; }
    @Override public double getSpeed(Particle p) { return p.ticks < 1 ? Math.random() * 2 : p.speed; }
    @Override public float getHue(Particle p) { return 60f / 360f; }
    @Override public float getSaturation(Particle p) { return 1; }
    @Override public float getBrightness(Particle p) { return 1; }
}
